### PR TITLE
Add NovaCustom as OEM for Clevo ADL-P laptops

### PR DESCRIPTION
**Pull request by arturkow2000**
Signed-off-by: Artur Kowalski <artur.kowalski@3mdeb.com>

### CODE REVIEW

**Review by miczyg1 — APPROVED** [review]

File: configs/config.novacustom_ns5x [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-CONFIG_BOARD_CLEVO_NS50MU=y
+CONFIG_MAINBOARD_SMBIOS_PRODUCT_NAME="NS50_70MU"
 CONFIG_CONSOLE_CBMEM_BUFFER_SIZE=0x20000
 CONFIG_EDK2_BOOT_TIMEOUT=2
-CONFIG_MAINBOARD_SMBIOS_PRODUCT_NAME="NS50_70MU"
 CONFIG_PCIEXP_HOTPLUG_BUSES=42
 CONFIG_PCIEXP_HOTPLUG_MEM=0xc200000
 CONFIG_PCIEXP_HOTPLUG_PREFETCH_MEM=0x1c000000
 CONFIG_EDK2_BOOTSPLASH_FILE="3rdparty/dasharo-blobs/novacustom/bootsplash.bmp"
+CONFIG_COREBOOT_ROMSIZE_KB_16384=y
```
Comment (miczyg1) [inline configs/config.novacustom_ns5x:15]: Isn't it 32MB?

File: configs/config.novacustom_ns5x [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
+# CONFIG_ONBOARD_VGA_IS_PRIMARY is not set
 CONFIG_VBOOT=y
-CONFIG_BOARD_CLEVO_NS50MU=y
+CONFIG_MAINBOARD_SMBIOS_PRODUCT_NAME="NS50_70MU"
 CONFIG_CONSOLE_CBMEM_BUFFER_SIZE=0x20000
 CONFIG_EDK2_BOOT_TIMEOUT=2
-CONFIG_MAINBOARD_SMBIOS_PRODUCT_NAME="NS50_70MU"
 CONFIG_PCIEXP_HOTPLUG_BUSES=42
 CONFIG_PCIEXP_HOTPLUG_MEM=0xc200000
 CONFIG_PCIEXP_HOTPLUG_PREFETCH_MEM=0x1c000000
```
Comment (miczyg1) [inline configs/config.novacustom_ns5x:13]: Do we need hotplug? What for?

File: configs/config.novacustom_ns5x [hunk 3]
```diff
@@ -1,17 +1,18 @@
 CONFIG_LOCALVERSION="v1.3.0"
-CONFIG_VENDOR_CLEVO=y
+CONFIG_VENDOR_NOVACUSTOM=y
+CONFIG_MAINBOARD_VERSION="2.2A"
 CONFIG_FMDFILE="src/mainboard/clevo/tgl-u/vboot-rwa.fmd"
-CONFIG_MAINBOARD_VENDOR="Notebook"
 # CONFIG_CONSOLE_SERIAL is not set
+# CONFIG_ONBOARD_VGA_IS_PRIMARY is not set
```
Comment (miczyg1) [inline configs/config.novacustom_ns5x:6]: Shouldn't onboard VGA be primary? One doesn;t boot laptop with dGPU turned on

**Review by miczyg1 — CHANGES_REQUESTED** [review]
Code looks good but not he config file itself

File: configs/config.novacustom_ns5x (hunk 2 above)
Comment (macpijan) [reply configs/config.novacustom_ns5x:13]: I think it's based on https://github.com/Dasharo/coreboot/blob/clevo/develop/configs/config.clevo_ns50mu and we had it here as well.

I know it does not answer **why** we have it there

File: configs/config.novacustom_ns5x (hunk 3 above)
Comment (arturkow2000) [reply configs/config.novacustom_ns5x]: Fixed

File: configs/config.novacustom_ns5x (hunk 2 above)
Comment (arturkow2000) [reply configs/config.novacustom_ns5x:12]: @miczyg1 @macpijan
Actually based on https://github.com/Dasharo/coreboot/blob/clevo/develop-adl-p/configs/config.clevo_ns70pu.

These configs are selected by `SOC_INTEL_ENABLE_USB4_PCIE_RESOURCES` https://github.com/Dasharo/coreboot/blob/clevo/develop-adl-p/src/soc/intel/alderlake/Kconfig#L213 which is enabled for all Alderlake Clevo platforms https://github.com/Dasharo/coreboot/blob/clevo/develop-adl-p/src/mainboard/clevo/adl-p/Kconfig#L22

File: src/mainboard/novacustom/adl-p/Kconfig.name [hunk 4]
```diff
@@ -0,0 +1,5 @@
+comment "Alder Lake P (2021)"
```
Comment (miczyg1) [inline src/mainboard/novacustom/adl-p/Kconfig.name:1]:
```suggestion
comment "Alder Lake P (2022)"
```

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**